PointPattern.py

```diff
@@ -0,0 +1,88 @@
+from .point import Point
+from . import analytics
+import random
+import numpy as np
+
+def PointPattern(object):
+    def __init__(self):
+        self.points = []
+
+    def average_nearest_neighbor_distance(self, mark=None):
+        return analytics.average_nearest_neighbor_distance(self.points, mark)
+
+    def add_point(self, point):
+        self.points.append(point)
+
+    def remove_point(self, index):
+        del(self.points[index])
+
+    def count_coincident_points(self):
+        count = 0
+        coincidnet_list = []
+
+        for i, point in enumerate(self.points):
+            for j, point2 in enumerate(self.points):
+                if i is j:
+                    continue
+                if j in coincident_list:
+                    continue
+                #Should use the magic method in point class
+                if point == point2:
+                    count += 1
+                    coincident_list.append(j)
+        return count
+
+    def list_marks(self):
+        marks = []
+
+        for point in self.points:
+            if point.mark is not None and point.mark not in marks:
+                marks.append(point.mark)
+
+    def return_subset(self, mark):
+        #creates a list of points that have the same mark as passed
+        return [i for i in self.points if i == mark]
```

Comment:
For a really fun approach, you could use `filter`.  Not required, but a more 'functional' approach.  The list comprehension you used is more Pythonic, so not real black and white rule for which is 'better'.